parsec/scheduling.c

```diff
@@ -185,8 +186,9 @@ int __parsec_execute( parsec_execution_stream_t* es,
                 /* Let's assume everything goes just fine */
                 task->status = PARSEC_TASK_STATUS_COMPLETE;
                 if(PARSEC_DEV_RECURSIVE >= tc->incarnations[chore_id].type) {
-                    /* accelerators count their own executed tasks */
-                    parsec_device_module_t *dev = parsec_mca_device_get(chore_id);
+                    /* accelerators count their own executed tasks so this is a
+                     * type DEV_RECURSIVE or DEV_CPU, they must be dev 0 and 1 */
+                    parsec_device_module_t *dev = parsec_mca_device_get(tc->incarnations[chore_id].type - PARSEC_DEV_CPU);
```

Comment:
That's not correct either... dev is a possible device for this chore, but not necessarily the device that we execute the task on.

If hook is a chore of type device, it will call `parsec_get_best_device`, and that's what will chose which device will actually execute the task...

@bosilca at some point you were working on a PR to move the get_best_device call before the hook. Do you still have this PR ? Can this solve this issue?

Reply:
this executes only if this is a CPU or RECURSIVE device, and we have only one of each.